source/dub/compilers/buildsettings.d

```diff
@@ -133,7 +133,9 @@ private:
 	// Append vals to arr without adding duplicates.
 	static void add(ref string[] arr, in string[] vals, bool noDuplicates = true)
 	{
-		arr ~= filterDuplicates(arr, vals, noDuplicates);
+		// vals might contain duplicates, add each val individually
+		foreach (val; vals)
+			arr ~= filterDuplicates(arr, [val], noDuplicates);
```

Comment:
Please add a unittest covering this case.

Reply:
Done